### PR TITLE
Improve Screen Variables Manager

### PR DESCRIPTION
**Pull request by caleeli**
## Issue & Reproduction Steps
Performance issues related to the data manager when typing on input text and updating calculated properties.

Expected behavior: 
Screen builder should perform faster.

Actual behavior: 
Performance issues related to the data manager when typing on input text and updating calculated properties.

## Solution
- Refactor of DataManajer.js and its dependencies to improve the performance of the screen builder

## How to Test

1. Import the attached process.
2. Run the process.
3. Open the assigned task
4. Type on the text area

[TestLesSchwabWarehouseCenter .zip](https://github.com/ProcessMaker/screen-builder/files/10340068/TestLesSchwabWarehouseCenter.zip)

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-6993
- https://processmaker.atlassian.net/browse/FOUR-7006

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-6993]: https://processmaker.atlassian.net/browse/FOUR-6993?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-7006]: https://processmaker.atlassian.net/browse/FOUR-7006?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/renderer/add-loop-row.vue [hunk 1]
```diff
@@ -20,14 +20,18 @@
...
+
 export default {
   props: {
     value: Array,
     config: Object,
     error: String,
   },
   methods: {
-    add() {
+    ...mapActions("globalErrorsModule", ["unlocked"]),
```
Comment (josechirivella) [inline src/components/renderer/add-loop-row.vue:32]: Can you change the function name from `unlocked` to something more relevant? Is hard to tell from the PR what the function is actually doing, or even reading from it in the code.

File: src/mixins/ScreenBase.js [hunk 2]
```diff
@@ -235,8 +303,27 @@ export default {
...
+        this.setValue(...args);
+      });
+    }, 210);
+    this.setValueDebounced = (...args) => {
+      debouncedValuesQueue.push(args);
+      setDebouncedValues();
+    };
+    this.setValueAsync = (name, value, object = this, defaults = object) =>
+      Promise.resolve().then(() => {
+        console.log(name, value, this.$el);
```
Comment (josechirivella) [inline src/mixins/ScreenBase.js:325]: remove console.log

**Review by josechirivella — CHANGES_REQUESTED** [review]

File: src/components/renderer/add-loop-row.vue (hunk 1 above)
Comment (caleeli) [reply src/components/renderer/add-loop-row.vue:32]: Seems this function is not required any more, I will review to remove this and the related ones

**Review by rodriquelca — APPROVED** [review]

File: src/mixins/ScreenBase.js (hunk 2 above)
Comment (caleeli) [reply src/mixins/ScreenBase.js]: fixed

**Review by josechirivella — APPROVED** [review]

**Review by rodriquelca — APPROVED** [review]

**Review by ryancooley — APPROVED** [review]

**Pull request merged**